File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Creating Custom SharePoint Service Applications/VB/WCF_CustomServiceApplication/IDayNamerContract.vb
Imports System.ServiceModel

Namespace Server

    'This interface defines the WCF service contract for
    'the custom service application. 
    <ServiceContract()> _
    Public Interface IDayNamerContract

        <OperationContract()> _
        Function TodayIs() As String

        <OperationContract()> _
        Function TodayAdd(ByVal daysToAdd As Integer) As String

    End Interface

End Namespace

File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Creating Custom SharePoint Service Applications/C#/WCF_CustomServiceApplication/IDayNamerContract.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ServiceModel;

namespace WCF_CustomServiceApplication.Server
{
    //This interface defines the WCF service contract for
    //the custom service application. 
    [ServiceContract]
    public interface IDayNamerContract
    {
        [OperationContract]
        string TodayIs();
        [OperationContract]
        string TodayAdd(int daysToAdd);
    }
}
